saleor/checkout/complete_checkout.py

```diff
@@ -559,7 +559,9 @@ def _create_order(
         additional_warehouse_lookup,
         check_reservations=is_reservation_enabled(site_settings),
         checkout_lines=[line.line for line in checkout_lines],
+        lock=False,
```

Comment:
I implemented suggested approach, with some exceptions:

1. Update of payment object is done out of transaction, within `_process_payment` function, but further in the code, in the second transaction, in case of any error we trigger `gateway.payment_refund_or_void` function, which updates the payments object back.
2. There is `validate_checkout_address` call out of transaction, which can update the address when is valid. Is it necessary to keep it within transaction?

Reply:
Checks are tricky, we may need to run some of them twice, before and after the payment call, each time with the checkout object locked, to make sure no one else is modifying the checkout while we turn it into an order. The stock check does not have to be repeated as long as we have a valid reservation.